### DISCUSSION TITLE
For jumping is there any way to get through the BoundingShape and stands on it when falling?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1063] [Q&A] [answered]
Hi:

For Castlevania like games, when characters jump usually they could get through the bounding shape
and stands on it when falling
Seems like it wouldn't work if we simply set the BoundingShape collidable? 
How should we do to achieve the desire effect?
For now I think it may work like remove CollidableComponent  when jumping upward and add it back when falling
but not sure this is the correct way or best practice?

Thanks

**Comment by AlmasB** [comment, marked as answer]
Hi,

CollidableComponent is only used for detecting / notifying collisions, but doesn't actually provide any rigid body physics (assuming you are using the PhysicsComponent / world combination).

There is no built-in solution for this, so a custom approach will probably be needed. You could detect the collision side by tracking previous player position and

1. if it's ⏫ then let the player pass
2. if it's ⏬ then set player's Y so it's on the platform and block the player's movement in Y axis.
